### PR TITLE
feat(MenuToggle): add plain with text variant

### LINKED ISSUE #6561: Menu toggle: add plain variant with text
Adds a new Menu toggle variant to be used for building composable menu demos. Core PR: https://github.com/patternfly/patternfly/pull/4491

### PR DESCRIPTION
**Pull request by nicolethoen**
**What**: Closes #6561

Added a new Menu Toggle example.

### CODE REVIEW

**Review by gabipodolnikova — APPROVED** [review]
LGTM!

**Review by mcarrano — COMMENTED** [review]
This looks good. @mcoker @mceledonia I think we covered this before, but I assume we are good with the idea that there is no hover effect on these plain variations?

**Review by wise-king-sullyman — APPROVED** [review]

**Comment by mceledonia** [comment]
What will it look like in terms of dev time if we add a hover state here to the caret? It will be subtle but consistent with our other icon action patterns. It would have the same colors/states, secondary text color (default) -> primary text color (hover) just to give it a small sense of feedback.

I know this isn't what we do on other dropdowns/selects, but it does feel like it needs some kind of state change since other similar components all do. 

That being said I am sort of on the fence here, if the juice isn't worth the squeeze I think the component will be fine without a hover state.

image


CC @mcoker @mcarrano

File: packages/react-core/src/components/MenuToggle/MenuToggle.tsx [hunk 1]
```diff
@@ -55,7 +55,8 @@ export class MenuToggleBase extends React.Component<MenuToggleProps> {
           styles.menuToggle,
           isExpanded && styles.modifiers.expanded,
           variant === 'primary' && styles.modifiers.primary,
-          isPlain && styles.modifiers.plain,
+          (isPlain || variant === 'plainText') && styles.modifiers.plain,
```
Comment (tlabaj) [inline packages/react-core/src/components/MenuToggle/MenuToggle.tsx:58]: Can you move the logic of checking the variant to line 39 where `isPlain` is set.

**Review by tlabaj — CHANGES_REQUESTED** [review]

**Comment by mcoker** [comment]
> What will it look like in terms of dev time if we add a hover state here to the caret?

@mceledonia pretty easy, we can try and get that in this release if you'd like. I'm assuming it should apply to all plain text toggles? Dropdown, options menu, select, etc?

**Comment by nicolethoen** [comment]
@mcoker 
If you update the hover styles, that shouldn't require an update on the react side, right? 
I opened a core issue and you can prioritize it however @mcarrano and @mceledonia see fit.

**Comment by nicolethoen** [comment]
Here is the new issue https://github.com/patternfly/patternfly/issues/4552

**Comment by mcarrano** [comment]
Thanks @nicolethoen . I will add this issue to the next milestone.

File: packages/react-core/src/components/MenuToggle/MenuToggle.tsx [hunk 2]
```diff
@@ -37,6 +37,8 @@ export class MenuToggleBase extends React.Component<MenuToggleProps> {
     const { children, className, icon, badge, isExpanded, isDisabled, variant, innerRef, ...props } = this.props;
 
     const isPlain = variant === 'plain';
+    const isPlainText = variant === 'plainText';
```
Comment (tlabaj) [inline packages/react-core/src/components/MenuToggle/MenuToggle.tsx:40]: you could combine these into one then line 58 would not have to change.
`const isPlain = variant === 'plain' || variant === 'plainText'`

Comment (nicolethoen) [reply packages/react-core/src/components/MenuToggle/MenuToggle.tsx:40]: Then i'd have to change the logic on lines 71-72. which I could do. But it's not true that if `plaintext` then `plain`. Plaintext also uses the expand/collapse arrow, unlike plain.

**Review by tlabaj — APPROVED** [review]

**Pull request merged**